src/conversation_engine/engine.py

```diff
@@ -105,9 +118,20 @@ def wait_for_user(self, target, missing_field):
         self._target = target
         self._missing_field = missing_field
 
-    def wait_for_robot(self):
-        rospy.loginfo("ConversationState: {old} -> {new}".format(old=self._state, new=ConversationState.WAIT_FOR_ROBOT))
+    def wait_for_robot(self, semantics):
```

Comment:
Docstring missing. Furthermore, I'm not sure how this works: the method is called `wait_for_robot` and the state is set accordingly but furthermore I don't see where there's actual waiting going on.

Reply:
Good point. The method is only called ```wait_for_robot``` because that is the name of the state we're transitioning to. I pondered for a second to name these ```transition_to_xxx``` but found that to long. I don't know a better name.

Reply:
`set_to_xxx` is more concise?

Reply:
Then you would get ```set_to_wait_for_robot```. The only public methods would then all start with ```set_to```, so I'd say to just the name of the state.

Reply:
But I'll agree this is confusing.